src/lunchselector/core.clj

```diff
@@ -8,54 +8,80 @@
   (let [params (:params request)
         token-response (lo/get-oauth-token (get params "code"))
-        access-token (get (cheshire/parse-string (:body token-response))
-                          "access_token")
-        user-details (lo/get-user-details access-token)]
+        access-token (:access_token (lu/parse-to-clj-map token-response))
+        user-details-resp (lo/get-user-details access-token)
+        user-details (lu/parse-to-clj-map user-details-resp)]
+    (lm/submit-users (:name user-details) (:email user-details))
     (-> (res/redirect "/")
-        (assoc :cookies {"user" {:value (:name (cheshire/parse-string (:body user-details) true))}}))))
+        (assoc :cookies {"username" {:value (:name user-details)}
+                         "useremail" {:value (:email user-details)}}))))
 
 (defn oauth [request]
+  "This page redirects to the Google OAuth URI for the OAuth dance"
   (res/redirect lo/oauth-redirect))
 
-(defn search [request]
+(defn home [request]
+  "Home Page. Displays a bunch of stuff"
   (let [cookies (:cookies request)
-        user (:value (get cookies "user"))]
+        user (:value (get cookies "username"))
+        email (:value (get cookies "useremail"))]
     (if (nil? user)
       (res/redirect "/oauth")
-      (res/response (lv/render-search user)))))
+      (do
+        (lm/submit-users user email)
+        (res/response (lv/render-home user email))))))
 
 (defn vote [request]
+  "Casts yours votes!"
   (let [params (:params request)
         votes (get params "restaurant")
         cookies (:cookies request)
-        user (:value (get cookies "user"))]
-    (lm/submit-votes user votes)
-    (res/response (lv/render-vote votes user))))
+        email (:value (get cookies "useremail"))
+        user (:value (get cookies "username"))]
+    (do  (lm/submit-online-restaurants votes)
+         (lm/submit-votes email votes)
+         (res/response (lv/render-vote votes user)))))
 
 (defn result [request]
-  (res/response (ldb/query)))
+  "Displays the votes for the day"
+  (res/response (lv/render-result
+                 (lm/votes-today))))
+
+(defn add-offline-restaurants [request]
+  "Adds offline restaurants to the DB and redirects to home page"
+  (let [params (:params request)
+        rest-name (get params "restaurant")
+        cookies (:cookies request)
+        email (:value (get cookies "useremail"))]
+    (do (lm/submit-restaurants {:name  rest-name :added-by email})
+        (res/redirect "/"))))
 
 (def handler
   (bidi/make-handler ["/" {
-                           "" search
+                           "" home
                            "restaurants" restaurants
                            "login" login
                            "oauth" oauth
                            "vote" vote
                            "result" result
+                           "add-offline-restaurants" add-offline-restaurants
                            }]))
 
 (def app
```

Comment:
I think it would be better to remove the routes and run-server functions into another namespace. `server` maybe. And this namespace could become a controller namespace exclusively.